runtime/src/anchor.rs

```diff
@@ -85,13 +97,20 @@ decl_module! {
             Ok(())
         }
 
-        pub fn commit(origin, anchor_id_preimage: T::Hash, doc_root: T::Hash, proof: T::Hash) -> Result {
-            // TODO make payable
+        pub fn commit(origin, anchor_id_preimage: T::Hash, doc_root: T::Hash, proof: T::Hash, eviction_date: T::Moment) -> Result {
             let who = ensure_signed(origin)?;
+            ensure!(<timestamp::Module<T>>::get() < eviction_date, "Not a valid storage date");
+
+            // validate the eviction date
+            let eviction_date_u64 = TryInto::<u64>::try_into(eviction_date)
+                .map_err(|_e| "Can not convert eviction date to u64")
+                .unwrap();
+            let storage_days = common::get_days_since_epoch(eviction_date_u64);
+            ensure!(Self::storage_max_days() >= storage_days, "Not a valid storage date");
```

Comment:
make it clear that max days is dynamic and relative to today (max days from now)